File: 240321/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.py
N = int(input())
cows = list(map(int,input().split()))

count = 0
for i in range(N):
    for j in range(i+1,N):
        if cows[i] <= cows[j]:
            for k in range(j+1,N):
                if cows[j] <= cows[k]:
                    count += 1

print(count)

File: 240404/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int N, count = 0, cows[100];

    scanf("%d",&N);
    for(int i = 0; i < N; i++)
        scanf("%d",&cows[i]);
    
    for(int i = 0; i < N; i++) {
        for(int j = i+1; j < N; j++) {
            if(cows[i] <= cows[j]) {
                for(int k = j+1; k < N; k++) {
                    if(cows[j] <= cows[k]) {
                        count++;
                    }
                }
            }
        }
    }

    printf("%d\n",count);

    return 0;
}